clippy_lints/src/non_send_fields_in_send_ty.rs

```diff
@@ -46,7 +50,7 @@ declare_clippy_lint! {
     #[clippy::version = "1.57.0"]
     pub NON_SEND_FIELDS_IN_SEND_TY,
     suspicious,
```

Comment:
The original example in the issue would be solved by this PR, but not the others shared later in the thread. I'm especially concerned about the intermediate struct example shared from `gtk-rs` which I know to be common. I believe it can be solvable with new rules, but for the time being I think it's better to stay in `nursery` to not disrupt those use cases. If you think it can be done as part of this PR, I'll add another commit to adjust the category soon.